src/App.tsx

```diff
@@ -1,61 +1,142 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { ToDo } from './types/todo';
+import { User } from './types/user';
+import { ChangeEvent, useState } from 'react';
+
+type TypeField = ChangeEvent<HTMLInputElement> | ChangeEvent<HTMLSelectElement>;
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const serverTodos: ToDo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const DEFAULT_VALUES = {
+  title: '',
+  userId: 0,
+  user: null,
+  id: 0,
+  completed: false,
+};
+
+const getNewTodoId = (todos: ToDo[]) => {
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<ToDo[]>(serverTodos);
+
+  const [formData, setFormData] = useState<ToDo>(DEFAULT_VALUES);
+  const [isTitleError, setIsTitleError] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isTitleError, setIsTitleError] = useState(false);
```
it's redundant to specify types for primitive